src/dune_rules/super_context.ml

```diff
@@ -288,10 +288,10 @@ let build_dir_is_vendored build_dir =
   in
   Option.value ~default:false opt
 
-let ocaml_flags t ~dir (x : Dune_file.Buildable.t) =
+let ocaml_flags t ~dir (f : Ocaml_flags.Spec.t) =
```

Comment:
Since `f` is often used for functions, it might be clearer to have this be

```suggestion
let ocaml_flags t ~dir (spec : Ocaml_flags.Spec.t) =
```